.github/workflows/python.yml

```diff
@@ -1,62 +1,125 @@
-name: Python CI Workflow
+name: Python CI/CD
 
 on:
   push:
   pull_request:
+  release:
+    types:
+      - published
 
 jobs:
 
-  build:
-    name: '[Python ${{ matrix.python-version }}]'
-    runs-on: ubuntu-20.04
+  build_sdist:
+    name: Build source distribution
+    runs-on: ubuntu-latest
     strategy:
       matrix:
         python-version:
-          - 3.6
           - 3.8
 
     steps:
 
       - uses: actions/checkout@master
       - run: git fetch --prune --unshallow
 
-      - name: Set up Python
+      - name: Install Python
         uses: actions/setup-python@v2
         with:
           python-version: ${{ matrix.python-version }}
 
-      - name: Install Python tools
-        run: pip install setuptools_scm
+      - name: Install sdist dependencies
+        run: pip install setuptools_scm cmake_build_extension
+
+      - name: Build sdist
+        run: python setup.py sdist
 
       - name: Install dependencies
         run: |
           sudo apt-get update
           sudo apt-get install -y \
-              build-essential coinor-libipopt-dev libeigen3-dev swig \
-              libxml2-dev libassimp-dev
-
-      - name: Create sdist
-        run: python setup.py sdist
+            build-essential \
+            coinor-libipopt-dev \
+            libeigen3-dev swig \
+            libxml2-dev \
+            libassimp-dev
 
       - name: Install sdist
         run: pip -v install dist/*.tar.gz
 
       - name: Test import
         run: python -c 'import idyntree.bindings'
 
-      - name: Remove sdist
-        run: |
-          rm -rf dist/
-          pip uninstall -y idyntree
+      - uses: actions/upload-artifact@v2
+        if: github.repository == 'robotology/idyntree'
+        with:
+          path: dist/*.tar.gz
+
+  build_wheels:
+    name: Build wheels on ${{ matrix.os }}
+    runs-on: ubuntu-latest
```

Comment:
Could it make sense to be explicit if we are using ubuntu-18.04 or ubuntu-20.04 . ubuntu-latest could make sense in a CI job, but for a deploy job it probably make sense to try to have explicit control on the version used.

Reply:
It does not matter much since `cibuildwheel` performs all the operations in a CentOS container. It could be also Ubuntu Hardy if you want (maybe I got a too old one xD).

Reply:
Ok, I meant that `cibuildwheel` itself could have a different behavior on Ubuntu 18.04 or 20.04, but if you think that is not a problem let's keep it like this.

Reply:
Since it's installed through pip and non the package manager, I confirm it should be ok.

Reply:
I'm giving a last pass to the changes, and I realized that in any case with different distros there no difference also because we configure a particular python version (3.8 here on bionic). `cibuildwheel` is then installed with that interpreter regardless of the distribution.